src/substrait-to-d3.js

```diff
@@ -0,0 +1,296 @@
+            row.appendChild(heading_1);
+            row.appendChild(heading_2);
+            propsTableHead.append(row);
+
+            for (let i = 0; i < nodeData.props.length; ++i) {
+                row = document.createElement('tr');
+                let data_1 = document.createElement('td');
+                let data_2 = document.createElement('td');
+                data_1.innerHTML = nodeData.props[i].name;
+                data_2.innerHTML = nodeData.props[i].value;
+                row.appendChild(data_1);
+                row.appendChild(data_2);
+                propsTableBody.appendChild(row);
+            }
+            node.appendChild(propsTable);
+        }
+
+        if (nodeData.schema.children.length) {
+            node.innerHTML += "<br>"
+            var childrenTable = document.createElement('table');
+            var childrenTableHead = document.createElement('thead');
+            var childrenTableBody = document.createElement('tbody');
+            var childrenTableCaption = document.createElement('caption');
+            childrenTableCaption.innerHTML = "Output Schema";
+            childrenTable.appendChild(childrenTableCaption);
+            childrenTable.appendChild(childrenTableHead);
+            childrenTable.appendChild(childrenTableBody);
+
+            let row = document.createElement("tr");
+            let heading_1 = document.createElement("th");
+            let heading_2 = document.createElement("th");
+            let heading_3 = document.createElement('th');
+            heading_1.innerHTML = "Name";
+            heading_2.innerHTML = "Type";
+            heading_3.innerHTML = "Nullability";
+            row.appendChild(heading_1);
+            row.appendChild(heading_2);
+            row.appendChild(heading_3);
+            childrenTableHead.append(row);
+
+            for (let i = 0; i < nodeData.schema.children.length; ++i) {
+                row = document.createElement('tr');
+                let data_1 = document.createElement('td');
+                let data_2 = document.createElement('td');
+                let data_3 = document.createElement('td');
+                data_1.innerHTML = nodeData.schema.children[i].name;
+                data_2.innerHTML = nodeData.schema.children[i].type;
+                data_3.innerHTML = nodeData.schema.children[i].nullability;
+                row.appendChild(data_1);
+                row.appendChild(data_2);
+                row.appendChild(data_3);
+                childrenTableBody.appendChild(row);
+            }
+            node.appendChild(childrenTable);
+        }
+
+    });
+
+    // specifying on tick function for the graph
+    force.on("tick", () => {
+        link.attr("d", (d) => {
+            return "M" + d.source.x + "," + d.source.y + "A0,0 0 0,1" + d.target.x + "," + d.target.y;
+        });
+        node.attr("transform", d => `translate(${d.x},${d.y})`);
+    });
+
+    // network drag simulation
+    function drag(network) {
```

Comment:
Dragging is cool.  Let's leave it in.  But maybe make it optional.  In a lot of cases we won't want the nodes to actually move around (though we still want to be able to click them).

Reply:
Thanks for the suggestion. Now, using a flag to enable dragging, with the default as true.